### PR TITLE
Integrate Cloud content generation with content_builder

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by GomathiselviS** [comment]
cc @GomathiselviS

File: plugins/action/generate_cloud_examples.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+from pathlib import Path
+from typing import Dict, List, Any, Union
+import re
+import ruamel.yaml
+import yaml
+from ansible.plugins.action import ActionBase
+
+
+class MissingDependency(Exception):
+    pass
```
Comment (justjais) [inline plugins/action/generate_cloud_examples.py:15]: can you give minor comments in both `MissingDependency` and `ContentInjectionFailure` exception classes

File: plugins/action/generate_cloud_examples.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    dependencies = [i for i in dependencies if not i.startswith("_")]
+    return sorted(list(set(dependencies)))
+
+
+def extract(
+    tasks: List[TaskType],
+    collection_name: str,
+    dont_look_up_vars: List[str],
+    task_selector: str = "name",
+) -> Dict[str, Any]:
```
Comment (justjais) [inline plugins/action/generate_cloud_examples.py:123]: Minor comment  to understand the functionality of respective fns.

File: plugins/action/generate_cloud_ignore_files.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
+import pathlib
+
+from ansible.plugins.action import ActionBase
+
+
+def refresh_ignore_files(target_dir):
+    module_utils_directory = pathlib.Path(target_dir +  "/plugins/module_utils") #target_dir / "plugins/module_utils"
```
Comment (justjais) [inline plugins/action/generate_cloud_ignore_files.py:7]: kindly remove the inline comment and move to a line above

File: plugins/action/generate_cloud_modules.py [hunk 4]
```diff
@@ -0,0 +1,1359 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright 2020 Red Hat
```
Comment (justjais) [inline plugins/action/generate_cloud_modules.py:3]: year update

File: plugins/action/generate_cloud_modules.py [hunk 5]
```diff
@@ -0,0 +1,1359 @@
...
+    UtilsBase,
+    jinja2_renderer,
+    get_module_added_ins,
+    get_module_from_config,
+    python_type,
+    camel_to_snake,
+    ignore_description,
+)
+from ansible_collections.ansible.content_builder.plugins.plugin_utils.cloud_utils.generator import generate_documentation
+
```
Comment (justjais) [inline plugins/action/generate_cloud_modules.py:38]: can we split this into vmware and aws specific and move the common functionalities to different .py file and then import common functionalities to be used by vmware and aws specific file to make this more readable and maintainable

**Review by justjais — COMMENTED** [review]
cosmetic comments

File: plugins/action/generate_cloud_modules.py (hunk 5 above)
Comment (GomathiselviS) [reply plugins/action/generate_cloud_modules.py:38]: The generate_documentaion imported here is aws specific. For vmware, the generation is straightforward and is done directly by a function in this file. I can add relevant comments before the import.

**Review by NilashishC — CHANGES_REQUESTED** [review]
@GomathiselviS could you please update the README with examples on how to scaffold this new plugin type? And also add a section (like security ones), that explain what the newly added keys do?

File: changelogs/fragments/cloud_content_action_plugins.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
```
Comment (NilashishC) [inline changelogs/fragments/cloud_content_action_plugins.yaml:2]: Can we remove all the intermediate changelogs, and add just one (minor_changes) reflecting the newly added scaffolding capabilities for cloud_content?

**Review by NilashishC — APPROVED** [review]

**Review by jillr — APPROVED** [review]

**Pull request merged**